tests/scripts/thread-cert/Cert_5_3_08_ChildAddressSet.py

```diff
@@ -30,81 +30,164 @@
+        self.sniffer.start()
 
     def tearDown(self):
+        self.sniffer.stop()
+        del self.sniffer
+
         for node in list(self.nodes.values()):
             node.stop()
         del self.nodes
 
     def test(self):
-        self.nodes[LEADER].start()
-        self.nodes[LEADER].set_state('leader')
-        self.assertEqual(self.nodes[LEADER].get_state(), 'leader')
-
-        self.nodes[ED1].start()
-        time.sleep(5)
-        self.assertEqual(self.nodes[ED1].get_state(), 'child')
-
-        self.nodes[ED2].start()
-        time.sleep(5)
-        self.assertEqual(self.nodes[ED2].get_state(), 'child')
+        self.nodes[DUT_LEADER].start()
+        self.nodes[DUT_LEADER].set_state('leader')
+        self.assertEqual(self.nodes[DUT_LEADER].get_state(), 'leader')
 
-        self.nodes[ED3].start()
+        self.nodes[BR].start()
         time.sleep(5)
-        self.assertEqual(self.nodes[ED3].get_state(), 'child')
-
-        self.nodes[ED4].start()
-        time.sleep(5)
-        self.assertEqual(self.nodes[ED4].get_state(), 'child')
-
-        for i in range(2,6):
-            addrs = self.nodes[i].get_addrs()
-            for addr in addrs:
-                if addr[0:4] != 'fe80':
-                    self.assertTrue(self.nodes[LEADER].ping(addr))
+        self.assertEqual(self.nodes[BR].get_state(), 'router')
+
+        # 1 BR: Configure BR to be a DHCPv6 server
+        self.nodes[BR].add_prefix('2001::/64', 'pdros')
+        self.nodes[BR].add_prefix('2002::/64', 'pdros')
+        self.nodes[BR].add_prefix('2003::/64', 'pdros')
+        self.nodes[BR].register_netdata()
+
+        # Set lowpan context of sniffer
+        self.sniffer.set_lowpan_context(1, '2001::/64')
+        self.sniffer.set_lowpan_context(2, '2002::/64')
+        self.sniffer.set_lowpan_context(3, '2003::/64')
+
+        # 2 DUT_LEADER: Verify LEADER sent an Advertisement
+        leader_messages = self.sniffer.get_messages_sent_by(DUT_LEADER)
+        msg = leader_messages.next_mle_message(mle.CommandType.ADVERTISEMENT)
+        command.check_mle_advertisement(msg)
+
+        # 3 MED1, MED2: MED1 and MED2 attach to DUT_LEADER
+        for i in MTDS:
+            self.nodes[i].start()
+            time.sleep(5)
+            self.assertEqual(self.nodes[i].get_state(), 'child')
+
+        # 4 MED1: MED1 send an ICMPv6 Echo Request to the MED2 ML-EID
+        med2_ml_eid = self.nodes[MED2].get_ip6_address(config.ADDRESS_TYPE.ML_EID)
+        self.assertTrue(med2_ml_eid != None)
+        self.assertTrue(self.nodes[MED1].ping(med2_ml_eid))
+
+        # Verify DUT_LEADER didn't generate an Address Query Request
+        leader_messages = self.sniffer.get_messages_sent_by(DUT_LEADER)
+        msg = leader_messages.next_coap_message('0.02', '/a/aq', False)
+        assert msg is None, "Error: The DUT_LEADER sent an unexpected Address Query Request"
+
+        # Wait for sniffer got packets
+        time.sleep(1)
```

Comment:
I think above ping pass in Line 117 already indicates ping reply happened. Not sure but Is this 1s necessary?

Reply:
The ping only check the final output result. There is about 1ms delay after the ping get its output result and then the sniffer get the Echo Reply message. Add 1s delay in Line 117 to make sure the sniffer has got the Echo Reply message.